x-pack/plugins/profiling/server/routes/setup.ts

```diff
@@ -86,11 +89,31 @@ export function registerSetupRoute({
           });
         }
 
-        state.data.available = await hasProfilingData({
-          ...setupOptions,
-          client: clientWithProfilingAuth,
-        });
-        if (state.data.available) {
+        const viewerUsersVerifyFunctions = [
```

Comment:
@jbcrail what do you think of this? Before we only checked if there was profiling data, but that is not enough for 8.9, as we need to verify other stuff too.

Reply:
It makes sense to me based on the discussion in Slack. The validation checks are in order of increasing higher user privileges.

I have two suggestions:
1. rename variables for easier reading
2. merge the partial states from the viewer into the partial states for the admin